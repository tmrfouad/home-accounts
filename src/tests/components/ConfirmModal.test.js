import React from 'react';
import ConfirmModal from '../../components/ConfirmModal';
import { shallow } from 'enzyme';

let isOpen, onModalClose, onModalOk, messageTitle, messageBody, wrapper;
beforeEach(() => {
  isOpen = true;
  onModalClose = jest.fn();
  onModalOk = jest.fn();
  messageTitle = 'Test Message Title';
  messageBody = 'Test message body';
  wrapper = shallow(
    <ConfirmModal
      isOpen={isOpen}
      onModalClose={onModalClose}
      onModalOk={onModalOk}
      messageTitle={messageTitle}
      messageBody={messageBody}
    />
  );
});

test('should render ConfirmModal correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onModalOk on ok button click', () => {
  wrapper
    .find('button')
    .at(0)
    .simulate('click');
  expect(onModalOk).toHaveBeenCalled();
});

test('should handle onModalClose on cancel button click', () => {
  wrapper
    .find('button')
    .at(1)
    .simulate('click');
  expect(onModalClose).toHaveBeenCalled();
});

test('should handle onModalClose when onRequestClose is called', () => {
  wrapper.find('#modal').prop('onRequestClose')();
  expect(onModalClose).toHaveBeenCalled();
});
