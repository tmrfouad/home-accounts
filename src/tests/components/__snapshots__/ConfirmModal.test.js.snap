// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConfirmModal correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Selected Option"
  id="modal"
  isOpen={true}
  onRequestClose={[MockFunction]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal__title"
  >
    Test Message Title
  </h3>
  <p
    className="modal__body"
  >
    Test message body
  </p>
  <button
    className="button button--secondary"
    onClick={[MockFunction]}
  >
    OK
  </button>
  <button
    className="button button--secondary"
    onClick={[MockFunction]}
  >
    Cancel
  </button>
</Modal>
`;
